node/modules/storageminer_idxprov.go

```diff
@@ -77,7 +77,6 @@ func IndexProvider(cfg config.IndexProviderConfig) func(params IdxProv, marketHo
 			ma := address.Address(maddr)
 			opts = append(opts,
 				engine.WithPublisherKind(engine.DataTransferPublisher),
-				engine.WithDataTransfer(dt),
```

Comment:
@masih so... what's this mean for users?

Reply:
not much, pull out whatever you want be cause this whole thing, and its dependency, is going when we land https://github.com/filecoin-project/lotus/pull/12005 :soon:

Reply:
Ooh. I didn't see that. I'll close this and wait.

Reply:
The default transport for syncing ad chains used to be data transfer a long long time ago. Now it is moved to use HTTP over libp2p as default with data transfer support being entirely deprecated and eventually removed.

Please feel free to remove all things data transfer related in the context of IPNI. Indexer servers are already upgraded to support the latest.

Reply:
> `engine.WithPublisherKind(engine.DataTransferPublisher)`

The line above needs to be updated to use HTTP over libp2p by default. Otherwise removing `WithDataTransfer` would cause all sorts of issues since index-provider engine would try to instantiate its own, which will cause a protocol ID registration conflict since every new instance of data transfer does that IIRC.